code/datums/components/crafting/guncrafting.dm

```diff
@@ -1,22 +1,165 @@
+	gun_suitable = FALSE
+	//A bit weak and a bit fast
+	added_description = "<span class='info'>The bowstring is made of bamboo fiber. Close to standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/silkstring
 	name = "silkstring"
-	desc = "A long piece of Silk that looks like a cable coil."
-	icon = 'icons/obj/improvised.dmi'
+	desc = "A string made out of silk to be used in weapon crafting."
 	icon_state = "silkstring"
+	damage_multiplier = 1
+	speed_multiplier = 1
+	gun_suitable = FALSE
+	//Standard stuff, no debuffs
+	added_description = "<span class='info'>The bowstring is made of silkstring. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/leatherstring
+	name = "leather string"
+	desc = "A string made out of leather to be used in weapon crafting."
+	icon_state = "leatherstring"
+	damage_multiplier = 1
+	speed_multiplier = 1.2
+	gun_suitable = FALSE
+	//Slight debuff to speed
+	added_description = "<span class='info'>The bowstring is made of leather. Standard strength.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/sinewstring
+	name = "sinew string"
+	desc = "A bow string made out of sinew to be used in weapon crafting."
+	icon_state = "sinewstring"
+	damage_multiplier = 1.2
+	speed_multiplier = 1.2
+	gun_suitable = FALSE
+	//Buff to damage, debuff to speed
+	added_description = "<span class='info'>The bowstring is made of sinew. It looks pretty strong.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/energy_crystal //Not aviable ingame yet
+	name = "energy crystal"
+	desc = "An energy crystal made out of uranium used in the construction of energy weaponry. A warning lable reads 'Warning: Do Not Ingest'."
+	custom_materials = list(/datum/material/uranium = MINERAL_MATERIAL_AMOUNT * 1)
+	material_flags = MATERIAL_EFFECTS
+	icon_state = "crystal"
+	damage_multiplier = 1.5
+	speed_multiplier = 0.6
+	fire_sound = 'sound/weapons/edagger.ogg'
+	//Stronger, faster
+	added_description = "<span class='info'>The bowstring is made of pure energy. As robust as it gets.</span>"
+
+/obj/item/weaponcrafting/attachment/primary/energy_crystal/clockwork //Made specially for clockcult, unretreaveable for now
+	name = "clockwork magic crystal"
+	desc = "An energy crystal made out of magic!"
+	custom_materials = null
+	icon_state = "clock_crystal"
+	damage_multiplier = 1
+	speed_multiplier = 1
+	added_description = "<span class='info'>The bowstring is made of pure energy. Able to create its own arrows.</span>"
+	//Ideally this stupid thing would create the arrows on its own, but for now thats tied to the bow itself
+
+// SECONDARY //
+// Secondary attachments, think of attachments as having slots they fit in, these fit into the "second" slot
+
+/obj/item/weaponcrafting/attachment/secondary/bowfangs
+	name = "bow fangs"
+	desc = "Fangs that can be attached to a bow to make it more suitable for hand to hand combat. It decreases accuracy, however."
+	icon_state = "fangs"
+	force = 2
+	spread = 2
```

Comment:
```suggestion
```

I don't think an accuracy reduction is necessary. This is a pretty minor increase in melee power, even considering the bleed power.

Reply:
I'd rather increase the force then, if thats okay. I really want this to be a meaning full choice, what do you think?

Reply:
If you want to increase the force that would be fine too

When I initially wrote this I hadn't gotten to your bow frame force values yet, and having glanced at them to properly answer this, your bow frames have such low force that your attachments don't matter at all - your bows are going to be far too weak to be usable melee weapons.

I'll touch on this more later when I get around to the force values on the individual frames

Reply:
Handled by increasing force instead, not only here, but on bows also.